crates/oxc_semantic/src/jsdoc/builder.rs

```diff
@@ -125,6 +160,75 @@ impl<'a> JSDocBuilder<'a> {
     }
 }
 
+// As noted above, only certain nodes can have JSDoc comments.
+// But as many kinds as possible should be added, without affecting performance.
+//
+// It's a bit hard to explain, but theoretically the more outer ones should be listed.
+//
+// From a linter point of view, basically only declarations are needed.
+// Other kinds, such as statements, act as tie-breakers between them.
+#[rustfmt::skip]
+fn should_attach_jsdoc(kind: &AstKind) -> bool {
+    println!("kind = {}", kind.debug_name());
```

Comment:
perf caused by this I think 😅

Reply:
Ah, sorry! 🙇🏻  [0a57420](https://github.com/oxc-project/oxc/pull/2623/commits/0a5742021b811e8fe9494591d74375b58bfc9b1a)